node/api/logs.go

```diff
@@ -33,10 +35,71 @@ type ContainerLogsHandlerFunc func(ctx context.Context, namespace, podName, cont
 // ContainerLogOpts are used to pass along options to be set on the container
 // log stream.
 type ContainerLogOpts struct {
-	Tail       int
-	Since      time.Duration
-	LimitBytes int
-	Timestamps bool
+	Tail         int
+	LimitBytes   int
+	Timestamps   bool
+	Follow       bool
+	Previous     bool
+	SinceSeconds int
+	SinceTime    time.Time
+}
+
+func parseLogOptions(q url.Values) (opts ContainerLogOpts, err error) {
+	if tailLines := q.Get("tailLines"); tailLines != "" {
+		opts.Tail, err = strconv.Atoi(tailLines)
+		if err != nil {
+			return opts, errdefs.AsInvalidInput(errors.Wrap(err, "could not parse \"tailLines\""))
+		}
+		if opts.Tail < 0 {
+			return opts, errdefs.InvalidInput(fmt.Sprintf("\"tailLines\" is %d", opts.Tail))
```

Comment:
Can you use `InvalidInputf` instead of `fmt.Sprintf`?

Reply:
There are a few of these.

Reply:
Sure, let me fix this :+1:

Reply:
@cpuguy83 done.